docs/040_pepr-tutorials/020_create-pepr-dashboard.md

```diff
@@ -0,0 +1,154 @@
+# Tutorial - Create a Pepr Dashboard
+
+## Introduction
+
+This tutorial will walk you through the process of creating a dashboard to display your Pepr metrics. This dashboard will present data such as the number of validation requests processed, the number of mutation requests that were allowed, the number of errors that were processed, the number of alerts that were processed, the status of the Pepr pods, and the scrape duration of the Pepr pods. This dashboard will be created using [Grafana](https://grafana.com/). The dashboard will display data from [Prometheus](https://prometheus.io/), which is a monitoring system that Pepr uses to collect metrics.
+
+This tutorial is not intended for production, but instead is intended to show how to quickly scrape Pepr metrics. The [kube prometheus stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack) provides a more production suitable way of deploying Prometheus in prod.
+
+An example of what the dashboard will look like is shown below:
+
+![Pepr Dashboard](../../dashboards/pepr-dashboard-screenshot.png)
+
+***Note:*** *The dashboard shown above is an example of what the dashboard will look like. The dashboard will be populated with data from your Pepr instance.*
+
+## Prerequisites
+
+1. Have a Pepr module created and deployed. If you do not have a Pepr module created and deployed, follow the [Create a Pepr Module](010_create-pepr-module.md) tutorial.
+2. Have a local Kubernetes cluster running. If you do not have a local Kubernetes cluster running, follow the [Create a Pepr Module](010_create-pepr-module.md) tutorial.
+
+## Steps
+
+### **Create a Prometheus deployment**
+
+#### Apply the CRDs to your cluster
+
+    ```bash
+    kubectl create -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/main/bundle.yaml
+    ```
+
+    This command creates in the `default` namespace the necessary Custom Resource Definitions that the Prometheus Operator will use.
+
+#### Apply the [Prometheus Operator deployment](../../dashboards/prometheus-operator.yaml) to your cluster
+
+    ```bash
+    kubectl apply -f prometheus-operator.yaml
+    ```
```

Comment:
```suggestion
```yaml
kubectl apply -f -<<EOF
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: prometheus-operator
rules:
  - apiGroups:
      - ""
    resources:
      - nodes
      - services
      - endpoints
      - pods
    verbs:
      - get
      - list
      - watch
---
apiVersion: v1
kind: ServiceAccount
metadata:
  name: prometheus-operator
  namespace: default
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: prometheus-operator
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: prometheus-operator
subjects:
  - kind: ServiceAccount
    name: prometheus-operator
    namespace: default
---
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: admission
spec:
  selector:
    matchLabels:
      pepr.dev/controller: admission
  namespaceSelector:
    matchNames:
    - pepr-system
  endpoints:
  - targetPort: 3000
    scheme: https
    tlsConfig:
      insecureSkipVerify: true
---
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: watcher
  namespace: pepr-system
spec:
  selector:
    matchLabels:
      pepr.dev/controller: watcher
  namespaceSelector:
    matchNames:
    - pepr-system
  endpoints:
  - targetPort: 3000
    scheme: https
    tlsConfig:
      insecureSkipVerify: true
 EOF
```
```

Reply:
As a consumer of this tutorial, I was to see what exactly I am applying without having to open up the file. Due to this, the flow is easier to understand IMO  if you are applying the body of the file instead of a file location